### ISSUE TITLE
HTML rendered default "Minimal Book Example" produces chaotic TOC

### ISSUE THREAD

**Issue by WilDoane** [rstudio/bookdown#1224]
After creating a new project from the bookdown template and knitting it, the resulting HTML has in section 1 the correct TOC:

![image](https://user-images.githubusercontent.com/550643/128575181-5a7deb67-320d-46c3-b5ab-e52eddf1118b.png)

but if you click thru to section 2 (or any other section) the TOC is misordered (note the placement of Methods):

![image](https://user-images.githubusercontent.com/550643/128575156-bae91e12-3a41-46db-8ec2-bd01bf9ddcfc.png)

```
> xfun::session_info('bookdown')
R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 11.5.1, RStudio 2021.8.0.294.0

Locale: en_US.UTF-8 / en_US.UTF-8 / en_US.UTF-8 / C / en_US.UTF-8 / en_US.UTF-8

Package version:
  base64enc_0.1.3   bookdown_0.22.17  digest_0.6.27     evaluate_0.14     glue_1.4.2       
  graphics_4.1.0    grDevices_4.1.0   highr_0.9         htmltools_0.5.1.1 jquerylib_0.1.4  
  jsonlite_1.7.2    knitr_1.33        magrittr_2.0.1    markdown_1.1      methods_4.1.0    
  mime_0.11         rlang_0.4.11      rmarkdown_2.10    stats_4.1.0       stringi_1.7.3    
  stringr_1.4.0     tinytex_0.33      tools_4.1.0       utils_4.1.0       xfun_0.25        
  yaml_2.2.1  
```

## Checklist

When filing a _bug report_, please check the boxes below to confirm that you have provided us with the information we need. Have you:

- [x] [formatted your issue](https://yihui.org/issue/#please-format-your-issue-correctly) so it is easier for us to read?

- [ ] included a minimal, self-contained, and reproducible example?

- [x] pasted the output from `xfun::session_info('bookdown')` in your issue?

- [x] upgraded all your packages to their latest versions (including your versions of R, the RStudio IDE, and relevant R packages)?

- [x] installed and tested your bug with the development version of the bookdown package using `remotes::install_github("rstudio/bookdown")` ?

**Comment by cderv** [comment]
We have just released **bookdown** 0.23 a few days ago: https://github.com/rstudio/bookdown/releases/tag/v0.23

It contains new project template. Can you try with this new template ? 

This should work ok. Thanks!

**Comment by WilDoane** [comment]
As far as I can tell, 2.3 appears to resolve this issue in my environment.
